pkg/git/checkout_test.go

```diff
@@ -0,0 +1,36 @@
+package git
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"testing"
+)
+
+func TestCheckoutTagSemVer_Checkout(t *testing.T) {
+	tag := CheckoutTag{
+		tag: "v1.7.0",
+	}
+	tmpDir, _ := ioutil.TempDir("", "test")
+	defer os.RemoveAll(tmpDir)
+
+	cTag, _, err := tag.Checkout(context.TODO(), tmpDir, "https://github.com/projectcontour/contour", nil)
```

Comment:
Maybe better to mock this than to rely on a third party repository?

Reply:
We can do this in a followup PR